Lib/tty.py

```diff
@@ -15,22 +24,81 @@
 OSPEED = 5
 CC = 6
 
-def setraw(fd, when=TCSAFLUSH):
-    """Put terminal into a raw mode."""
-    mode = tcgetattr(fd)
-    mode[IFLAG] = mode[IFLAG] & ~(BRKINT | ICRNL | INPCK | ISTRIP | IXON)
-    mode[OFLAG] = mode[OFLAG] & ~(OPOST)
-    mode[CFLAG] = mode[CFLAG] & ~(CSIZE | PARENB)
-    mode[CFLAG] = mode[CFLAG] | CS8
-    mode[LFLAG] = mode[LFLAG] & ~(ECHO | ICANON | IEXTEN | ISIG)
+try:
+    CHECK_WINSZ = TIOCGWINSZ
+    CHECK_WINSZ = TIOCSWINSZ
+    HAVE_WINSZ = True
+
+    def getwinsz(fd):
+        """Gets window size of tty.
+        If fd is not a descriptor of
+        a tty, then OSError is raised."""
+        s = pack("HHHH", 0, 0, 0, 0)
+        return ioctl(fd, TIOCGWINSZ, s)
+
+    def setwinsz(fd, winsz):
+        """Sets window size of tty.
+        If fd is not a descriptor of
+        a tty, then OSError is raised."""
+        ioctl(fd, TIOCSWINSZ, winsz)
+except NameError:
```

Comment:
Explicit check for `hasattr(termios, "TIOCSWINSZ")` or something like this is better than `except NameError`.

Reply:
It did feel like there had to be a better way of doing this. I did not like the `except NameError` either. Thank you so much for this suggestion.

Reply:
I have one concern. To be able to use `hasattr(termios, "TIOCSWINSZ")`, I think I need to `import termios`. Am I correct about this? Since we already have `from termios import *`, which cannot be removed because of backward compatibility, is it good practice to also have an `import termios` there just for that single purpose? I was thinking that we could use

```
try:
   from termios import TIOCGWINSZ, TIOCSWINSZ
   ...
except ImportError:
   ...
```

which seems to be an explicit check as well.